lib/src/mistralai_client_dart_base.dart

```diff
@@ -2,59 +2,71 @@ import 'dart:convert';
 
 import 'package:http/http.dart' as http;
 import 'package:http/retry.dart' as retry;
+import 'package:mistralai_client_dart/src/exceptions.dart';
 import 'package:mistralai_client_dart/src/models/models.dart';
 
-class _MistralAPIEndpoints {
-  static const String listModels = '/v1/models';
-  static const String embeddings = '/v1/embeddings';
-}
+part 'network/request.dart';
+part 'models/mappers.dart';
+part 'network/url_tools.dart';
+
+// FIXME(lgawron): add tests for all requests
+// TODO(lgawron): add more documentation
 
+/// Mistral AI Client
 class MistralAIClient {
+  /// Creates a new instance of [MistralAIClient].
+  ///
+  /// [apiKey] is required to authenticate requests.
+  ///
+  /// [baseUrl] is the base url for all requests.
+  ///
+  /// [timeout] is the maximum wait time for response.
+  ///
+  /// [maxRetries] is the maximum number of retries for a single request.
+  /// Client uses [retry.RetryClient] to retry requests.
+  ///
+  /// [apiUrlFactory] allows to override default [MistralAIUrlFactory].
+  /// It can be used to override default paths for requests when Mistral AI API
+  /// changes.
```

Comment:
In theory when the API changes we should update our SDK ASAP, so not sure if this documentation makes sense 🤔  Maybe it would be better to phrase it like this:
```suggestion
  /// It can be used to override default paths for requests to Mistral AI API.
  /// It could be useful when trying to use a non-stable version of the API.
```

Reply:
Resolved in [5019cfd](https://github.com/nomtek/mistralai_client_dart/pull/18/commits/5019cfda4bf3fa7a65970d922744f964f2d956a9)